From 959ea0713c70a3311303d02c214afc319a25868d Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Fri, 11 Apr 2025 13:29:15 -0400
Subject: [PATCH] vibrator: fixup service declaration for loading effects

The previous declaration was resulting in non-working vibrator effects
on waffle due to the perms being set too late and the service running
too soon at the same time.

[YAAP:] original is
https://review.lineageos.org/c/LineageOS/android_hardware_oplus/+/426367

However this solved the same issue on waffle using common service
---
 aidl/vendor.qti.hardware.vibrator.service.rc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/aidl/vendor.qti.hardware.vibrator.service.rc b/aidl/vendor.qti.hardware.vibrator.service.rc
index 718a12d..93f7cd2 100644
--- a/aidl/vendor.qti.hardware.vibrator.service.rc
+++ b/aidl/vendor.qti.hardware.vibrator.service.rc
@@ -1,9 +1,9 @@
-on late-init
+on boot
     chown system system /sys/class/qcom-haptics/primitive_duration
     chmod 0600 /sys/class/qcom-haptics/primitive_duration
 
 service vendor.qti.vibrator /vendor/bin/hw/vendor.qti.hardware.vibrator.service
-    class hal
+    class late_start
     user system
     group system input
 
